### PR TITLE
Supervisor disk usage more info

### LINKED ISSUE #7241: FR: Add absolute total/available space to the System Metrics panel in Supervisor->System
**Home Assistant release with the issue:**

0.116.0b3

**Operating environment (HassOS/Generic):**

HassOS in KVM

**Supervisor logs:**

**Description of problem:**

The new System Metrics panel has a lot of great info, including the storage space that's used.  It currently only shows a percentage, and that can make it difficult to support or triage issues without knowing how big the storage device actually is.  Adding a "used/total" along with the percentage would help identify people who are using small (sometimes way too small) SD cards, or catch misconfigurations in a VM when the image hasn't been expanded.  90% of an 8/16/32GB card tells a different story than 90% of a 100+GB partition, and it's not obvious from the current display which situation we're in.

image

### PR DESCRIPTION
**Pull request by maykar**
## Proposed change

Adds more info about used and available disk space on hover of disk usage bar in supervisor.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #7241 

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/system/hassio-system-metrics.ts [hunk 1]
```diff
@@ -77,7 +92,11 @@ class HassioSystemMetrics extends LitElement {
     this._loadData();
   }
 
-  private _renderMetric(description: string, value: number): TemplateResult {
+  private _renderMetric(
+    description: string,
+    value: number,
+    title: string
```
Comment (ludeeus) [inline hassio/src/system/hassio-system-metrics.ts:98]: let's use tooltip instead title where you have that added, and make it optional

File: hassio/src/system/hassio-system-metrics.ts [hunk 2]
```diff
@@ -38,35 +39,49 @@ class HassioSystemMetrics extends LitElement {
...
       },
       {
         description: "Supervisor CPU Usage",
         value: this._supervisorMetrics?.cpu_percent,
+        title: "",
       },
       {
         description: "Supervisor RAM Usage",
         value: this._supervisorMetrics?.memory_percent,
+        title: "",
```
Comment (ludeeus) [inline hassio/src/system/hassio-system-metrics.ts:67]: Remove blank values

File: hassio/src/system/hassio-system-metrics.ts [hunk 3]
```diff
@@ -38,35 +39,49 @@ class HassioSystemMetrics extends LitElement {
   @internalProperty() private _coreMetrics?: HassioStats;
 
   protected render(): TemplateResult | void {
-    const usedSpace = this._getUsedSpace(this.hostInfo);
+    const host = this.hostInfo
+    const usedSpace = this._getUsedSpace(host);
+    const usedSpaceTitle =
```
Comment (ludeeus) [inline hassio/src/system/hassio-system-metrics.ts:44]: Use template string

**Review by ludeeus — CHANGES_REQUESTED** [review]

File: hassio/src/system/hassio-system-metrics.ts [hunk 4]
```diff
@@ -38,7 +39,14 @@ class HassioSystemMetrics extends LitElement {
   @internalProperty() private _coreMetrics?: HassioStats;
 
   protected render(): TemplateResult | void {
-    const usedSpace = this._getUsedSpace(this.hostInfo);
+    const host = this.hostInfo;
```
Comment (ludeeus) [inline hassio/src/system/hassio-system-metrics.ts:42]: You don't need this

File: hassio/src/system/hassio-system-metrics.ts [hunk 5]
```diff
@@ -38,7 +39,14 @@ class HassioSystemMetrics extends LitElement {
...
+    const usedSpace = this._getUsedSpace(host);
+    const usedSpaceTip = `${bytesToString(
```
Comment (ludeeus) [inline hassio/src/system/hassio-system-metrics.ts:44]: You can add this direcly in the metrics object, no need to make a const of it

File: hassio/src/system/hassio-system-metrics.ts [hunk 6]
```diff
@@ -38,7 +39,14 @@ class HassioSystemMetrics extends LitElement {
...
+      host.disk_used * 1e6
+    )}/${bytesToString(host.disk_total * 1e6)}`;
+    const coreRamTip = `${bytesToString(
+      this._coreMetrics?.memory_usage
+    )}/${bytesToString(this._coreMetrics?.memory_limit)}`;
```
Comment (ludeeus) [inline hassio/src/system/hassio-system-metrics.ts:49]: This displays as MB but should be GB on my device

**Review by ludeeus — CHANGES_REQUESTED** [review]
Supervisor RAM is missing tip

File: hassio/src/system/hassio-system-metrics.ts (hunk 5 above)
Comment (ludeeus) [reply hassio/src/system/hassio-system-metrics.ts:44]: This displays as MB but should be GB on my device

File: hassio/src/system/hassio-system-metrics.ts [hunk 7]
```diff
@@ -55,18 +58,28 @@ class HassioSystemMetrics extends LitElement {
...
 
     return html`
       <ha-card header="System Metrics">
         <div class="card-content">
           ${metrics.map((metric) =>
-            this._renderMetric(metric.description, metric.value ?? 0)
+            this._renderMetric(
+              metric.description,
+              metric.value ?? 0,
+              metric.tooltip ?? undefined
```
Comment (ludeeus) [inline hassio/src/system/hassio-system-metrics.ts:81]:
```suggestion
              metric.tooltip
```

**Review by ludeeus — APPROVED** [review]
:+1:

File: src/util/calculate.ts [hunk 8]
```diff
@@ -21,3 +21,12 @@ export const getValueInPercentage = (
...
 };
+
+export const bytesToString = (value = 0, decimals = 2): string => {
+    if (value === 0) return '0 Bytes';
+    const k = 1024;
+    decimals = decimals < 0 ? 0 : decimals;
+    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+    const i = Math.floor(Math.log(value) / Math.log(k));
+    return `${parseFloat((value / k ** i).toFixed(decimals))} ${sizes[i]}`;
+}
```
Comment (bramkragten) [inline src/util/calculate.ts:32]: Let's move this to its own file

Comment (maykar) [reply src/util/calculate.ts:32]: Generic file to be added to later like `src/util/convert.ts` or just it's own `src/util/bytes-to-string.ts`?

Comment (bramkragten) [reply src/util/calculate.ts:32]: it's own

**Review by ludeeus — APPROVED** [review]

**Pull request merged**